test/server/services/user.js

```diff
@@ -65,6 +68,65 @@ describe('User service tests', () => {
     });
   });
 
+  describe('Campaign User association', function() {
+    before((done) => {
+      this.userSaveParam = {
+        firstName: 'Jack',
+        lastName: 'J',
+        password: 'hatch1',
+        phoneNumber: '+144411144442',
+        email: 'Jack@gmail.com'
+      };
+
+      this.contactListParam = {
+        name: 'Millbrae Phone List 3'
+      };
+
+      this.campaignParams = {
+        name: 'testCampaign4',
+        title: 'Test4',
+        description: 'election',
+        status: 'draft'
+      };
+
+      this.scriptParams = {
+        name: 'Script Name 4',
+        body: 'Script Body 4',
+        description: 'Script Description 4'
+      };
+
+      this.campaignUserParam = {};
+
+      User.saveNewUser(this.userSaveParam)
+        .then((user) => {
+          this.campaignUserParam.id = user.attributes.id;
+          Script.saveNewScript(this.scriptParams)
+            .then((script) => {
+              this.campaignParams.script_id = script.attributes.id;
+              ContactList.saveNewContactList(this.contactListParam)
+                .then((contactList) => {
+                  this.campaignParams.contact_lists_id = contactList.attributes.id;
+                  Campaign.saveNewCampaign(this.campaignParams)
+                    .then((campaign) => {
+                      this.campaignUserParam.campaign_id = campaign.attributes.id;
+                      done();
+                    });
+                });
+            });
+        });
+    });
+
+
+    it('should add Campaign User association', (done) => {
+      User.addCampaignToUser(this.campaignUserParam)
+        .then((campaignUser) => {
+          expect(campaignUser.models[0].attributes.user_id).to.equal(3);
```

Comment:
Instead of 3 here, I'd write this.campaignUserParam.id and this.campaignUserParam.campaign_id